Remotion/Web/ClientScript/Scripts/ListMenu.ts

```diff
@@ -0,0 +1,315 @@
+    diagnosticMetadataForCommand: Dictionary<string | boolean>)
+  {
+    this._id = id;
+    this._category = category;
+    this._text = text;
+    this._icon = icon;
+    this._iconDisabled = iconDisabled;
+    this._requiredSelection = requiredSelection;
+    this._isDisabled = isDisabled;
+    this._href = href;
+    this._target = target;
+    this._diagnosticMetadata = diagnosticMetadata;
+    this._diagnosticMetadataForCommand = diagnosticMetadataForCommand;
+  }
+}
+
+class ListMenu
+{
+  private readonly _itemClassName: string = 'listMenuItem';
+  private readonly _itemFocusClassName: string = 'listMenuItemFocus';
+  private readonly _itemDisabledClassName: string = 'listMenuItemDisabled';
+  private readonly _requiredSelectionAny: number = 0;
+  private readonly _requiredSelectionExactlyOne: number = 1;
+  private readonly _requiredSelectionOneOrMore: number = 2;
+  private _listMenuInfos: Dictionary<ListMenuMenuInfo> = {};
+
+  public Initialize ($listMenu: JQuery): void
+  {
+    ArgumentUtility.CheckNotNullAndTypeIsJQuery("$listMenu", $listMenu);
+
+    $listMenu.keydown ((event) =>
+    {
+      this.OnKeyDown (event, $listMenu);
+    });
+  }
+
+  public AddMenuInfo (listMenu: Element, menuInfo: ListMenuMenuInfo)
+  {
+    this._listMenuInfos[listMenu.id] = menuInfo;
+  }
+
+  public Update (listMenu: Element, getSelectionCount: () => number)
+  {
+    var menuInfo = this._listMenuInfos[listMenu.id];
+    if (menuInfo == null)
+      return;
+
+    var itemInfos = menuInfo.ItemInfos;
+    var selectionCount = -1;
+    if (getSelectionCount != null)
+      selectionCount = getSelectionCount();
+
+    for (var i = 0; i < itemInfos.length; i++)
+    {
+      var itemInfo = itemInfos[i];
+      var isEnabled = true;
+      if (itemInfo.IsDisabled)
+      {
+        isEnabled = false;
+      }
+      else
+      {
+        if (itemInfo.RequiredSelection == this._requiredSelectionExactlyOne && selectionCount != 1)
+        {
+          isEnabled = false;
+        }
+        if (itemInfo.RequiredSelection == this._requiredSelectionOneOrMore && selectionCount < 1)
+        {
+          isEnabled = false;
+        }
+      }
+      var item = <Element>document.getElementById (itemInfo.ID);
+      var anchor = $(item).children().eq (0)[0] as HTMLAnchorElement;
+      var icon = $(anchor).children().eq (0)[0] as HTMLImageElement;
+      if (isEnabled)
+      {
+        if (icon != null && icon.nodeType == 1)
+          icon.src = itemInfo.Icon;
+        item.className = this._itemClassName;
+        if (itemInfo.Href != null)
+        {
+          if (itemInfo.Href.toLowerCase().indexOf ('javascript:') >= 0)
+          {
+            anchor.href = '#';
+            anchor.removeAttribute ('target');
+            anchor.setAttribute ('javascript', itemInfo.Href);
+            anchor.removeAttribute ('onclick');
+            anchor.onclick = () => { eval (<string>anchor.getAttribute('javascript')); return false; };
```

Comment:
1. `anchor.getAttribute('javascript')` should have a null check - bug?
2. I replaced `this.getAttribute(...)` with `anchor.getAttribute(...)` here. Was this the right decision?

Reply:
@iam-mholle 
1. No, that's only "helpful" if we worry that someone manipulates our DOM after the script has been initialized.
2. {{this}} was chosen because the click-handler is on the anchor, so anchor is {{this}}. I didn't want to capture the anchor in the handler. There's also historical issue where DOM/Handler+Handler/DOM dependencies make a cyclic dependency that cannot be collected.